tests/user_commands/tests.py

```diff
@@ -535,3 +536,10 @@ def test_is_ignored_path_false(self):
     def test_normalize_path_patterns_truncates_wildcard_base(self):
         expected = [os.path.normcase(p) for p in ["foo/bar", "bar/*/"]]
         self.assertEqual(normalize_path_patterns(["foo/bar/*", "bar/*/"]), expected)
+
+    @mock.patch("subprocess.run", side_effect=OSError)
```

Comment:
I think it would be a better approach to mock the result of `shutil.run`, something like:
```suggestion
    @mock.patch("django.core.management.utils.shutil.which", return_value="/foo.bar.baz")
```

Reply:
Then we would need to create a test for an existing file without proper perms to ensure PermissionError is tested.